contrib/vagrant/start.sh

```diff
@@ -601,9 +601,9 @@ elif [ -n "${PROVISION}" ]; then
     vagrant provision
 else
     vagrant up
-    if [ -n "${K8S}" ]; then
-		vagrant ssh k8s1 -- cat /home/vagrant/.kube/config | sed 's;server:.*:6443;server: https://k8s1:7443;g' > vagrant.kubeconfig
-		echo "Add '127.0.0.1 k8s1' to your /etc/hosts to use vagrant.kubeconfig file for kubectl"
-	fi
+    if [ "$?" -eq "0" -a -n "${K8S}" ]; then
```

Comment:
I think that a similar test should be applied to the `${K8S}` variable as its value may be set to `0` (to disable k8s), in which case one should not enter this `if` condition.

Reply:
Good point, done.

Reply:
Actually, I think `-n ${K8S}` was correct here. I'm now getting the following error when `K8S` is unset:
```
./contrib/vagrant/start.sh: line 607: [: : integer expression expected
```
`-n` also works when `K8S=0`; [it's `-z` that doesn't in that case](https://stackoverflow.com/a/3870055/6884590).

I'll open a PR to revert this specific change and will change other places that use `-z` in the script if trying `K8S=0` confirms that it's needed.